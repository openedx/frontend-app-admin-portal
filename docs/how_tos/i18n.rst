####################
React App i18n HOWTO
####################

This document has moved to the frontend-platform repo: https://github.com/openedx/frontend-platform/blob/master/docs/how_tos/i18n.rst
