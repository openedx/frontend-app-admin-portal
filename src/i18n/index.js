import { messages as paragonMessages } from '@edx/paragon';

export default [
  paragonMessages,
];
