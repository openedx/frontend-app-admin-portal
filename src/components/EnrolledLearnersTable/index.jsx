import React from 'react';

import { useIntl } from '@edx/frontend-platform/i18n';

import TableContainer from '../../containers/TableContainer';
import { i18nFormatTimestamp } from '../../utils';
import EnterpriseDataApiService from '../../data/services/EnterpriseDataApiService';

const EnrolledLearnersTable = () => {
  const intl = useIntl();

  const tableColumns = [
    {
      label: intl.formatMessage({
        id: 'admin.portal.lpr.enrolled.learners.table.user_email.column.heading',
        defaultMessage: 'Email',
        description: 'Column heading for the user email column in the enrolled learners table',
      }),
      key: 'user_email',
      columnSortable: true,
    },
    {
      label: intl.formatMessage({
        id: 'admin.portal.lpr.enrolled.learners.table.lms_user_created.column.heading',
        defaultMessage: 'Account Created',
        description: 'Column heading for the lms user created column in the enrolled learners table',
      }),
      key: 'lms_user_created',
      columnSortable: true,
    },
    {
      label: intl.formatMessage({
        id: 'admin.portal.lpr.enrolled.learners.table.enrollment_count.column.heading',
        defaultMessage: 'Total Course Enrollment Count',
        description: 'Column heading for the course enrollment count column in the enrolled learners table',
      }),
      key: 'enrollment_count',
      columnSortable: true,
    },
  ];

  const formatLearnerData = learners => learners.map(learner => ({
    ...learner,
    user_email: <span data-hj-suppress>{learner.user_email}</span>,
    lms_user_created: i18nFormatTimestamp({
      intl, timestamp: learner.lms_user_created,
    }),
  }));

  return (
    <TableContainer
      id="enrolled-learners"
      className="enrolled-learners"
      fetchMethod={EnterpriseDataApiService.fetchEnrolledLearners}
      columns={tableColumns}
      formatData={formatLearnerData}
      tableSortable
    />
  );
};

export default EnrolledLearnersTable;
