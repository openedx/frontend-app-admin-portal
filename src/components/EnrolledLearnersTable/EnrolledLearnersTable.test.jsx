import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { IntlProvider } from '@edx/frontend-platform/i18n';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';

import EnrolledLearnersTable from '.';

const mockStore = configureMockStore([thunk]);
const enterpriseId = 'test-enterprise';
const store = mockStore({
  portalConfiguration: {
    enterpriseId,
  },
  table: {
    'enrolled-learners': {
      data: {
        results: [],
        current_page: 1,
        num_pages: 1,
      },
      ordering: null,
      loading: false,
      error: null,
    },
  },
});

const EnrolledLearnersWrapper = props => (
  <MemoryRouter>
    <IntlProvider locale="en">
      <Provider store={store}>
        <EnrolledLearnersTable
          {...props}
        />
      </Provider>
    </IntlProvider>
  </MemoryRouter>
);

describe('EnrolledLearnersTable', () => {
  it('renders empty state correctly', () => {
    const tree = renderer
      .create((
        <EnrolledLearnersWrapper />
      ))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
