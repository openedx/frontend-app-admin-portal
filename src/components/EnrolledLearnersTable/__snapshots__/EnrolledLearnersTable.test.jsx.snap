// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnrolledLearnersTable renders empty state correctly 1`] = `
<div
  className="pgn__data-table-layout-wrapper"
>
  <div
    className="pgn__data-table-layout-main"
  >
    <div
      className="pgn__data-table-wrapper"
    >
      <div
        className="pgn__data-table-status-bar"
        data-testid="table-control-bar"
      >
        <div
          className="pgn__data-table-actions"
        >
          <div
            className="pgn__data-table-actions-left"
          >
            <div
              className="pgn__data-table-filters"
            >
              <div
                className="pgn__data-table-filters-breakout-filter"
              >
                <div
                  className="pgn__form-group"
                >
                  <label
                    className="pgn__form-label sr-only"
                    htmlFor="text-filter-label-header_userEmail-1"
                  >
                    Search email
                  </label>
                  <div
                    className="pgn__form-control-decorator-group"
                  >
                    <input
                      className="form-control"
                      id="text-filter-label-header_userEmail-1"
                      onBlur={[Function]}
                      onChange={[Function]}
                      placeholder="Search email"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="pgn__data-table-actions-right"
          >
            <div />
          </div>
        </div>
        <div
          className="pgn__data-table-status"
        >
          <div
            className="pgn__data-table-status-left"
          />
          <div
            className="pgn__data-table-toggle"
          />
        </div>
      </div>
      <div
        className="pgn__data-table-container is-loading"
        data-testid="data-table-container"
      >
        <div
          className="pgn__data-table-spinner"
          data-testid="data-table-spinner"
        >
          <div
            className="pgn__spinner spinner-border"
            role="status"
          >
            <span
              className="sr-only"
            >
              loading
            </span>
          </div>
        </div>
        <table
          className="pgn__data-table is-striped"
          role="table"
        >
          <thead>
            <tr
              role="row"
            >
              <th
                colSpan={1}
                onClick={[Function]}
                role="columnheader"
                style={
                  {
                    "cursor": "pointer",
                  }
                }
                title="Toggle SortBy"
              >
                <span
                  className="d-flex align-items-center"
                >
                  <span>
                    Email
                  </span>
                  <span
                    className="pgn__icon"
                    data-testid="arrow-drop-up-down"
                    style={
                      {
                        "opacity": 0.5,
                      }
                    }
                  >
                    <svg
                      aria-hidden={true}
                      fill="none"
                      focusable={false}
                      height={24}
                      role="img"
                      viewBox="0 0 24 24"
                      width={24}
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="m7 10 5-5 5 5H7ZM7 14l5 5 5-5H7Z"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                </span>
              </th>
              <th
                colSpan={1}
                onClick={[Function]}
                role="columnheader"
                style={
                  {
                    "cursor": "pointer",
                  }
                }
                title="Toggle SortBy"
              >
                <span
                  className="d-flex align-items-center"
                >
                  <span>
                    Account Created
                  </span>
                  <span
                    className="pgn__icon"
                    data-testid="arrow-drop-down"
                  >
                    <svg
                      aria-hidden={true}
                      fill="none"
                      focusable={false}
                      height={24}
                      role="img"
                      viewBox="0 0 24 24"
                      width={24}
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="m7 10 5 5 5-5H7Z"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                </span>
              </th>
              <th
                colSpan={1}
                onClick={[Function]}
                role="columnheader"
                style={
                  {
                    "cursor": "pointer",
                  }
                }
                title="Toggle SortBy"
              >
                <span
                  className="d-flex align-items-center"
                >
                  <span>
                    Total Course Enrollment Count
                  </span>
                  <span
                    className="pgn__icon"
                    data-testid="arrow-drop-up-down"
                    style={
                      {
                        "opacity": 0.5,
                      }
                    }
                  >
                    <svg
                      aria-hidden={true}
                      fill="none"
                      focusable={false}
                      height={24}
                      role="img"
                      viewBox="0 0 24 24"
                      width={24}
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="m7 10 5-5 5 5H7ZM7 14l5 5 5-5H7Z"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                </span>
              </th>
            </tr>
          </thead>
          <tbody
            role="rowgroup"
          />
        </table>
      </div>
      <div
        className="pgn__data-table-footer"
        data-testid="table-footer"
      >
        <nav
          aria-label="table pagination"
          className="pagination-minimal"
        >
          <div
            aria-atomic={true}
            aria-live="polite"
            aria-relevant="text"
            className="sr-only"
          >
            Page 1, Current Page, of -1
          </div>
          <ul
            className="pagination"
          >
            <li
              className="page-item disabled"
            >
              <button
                aria-label="Previous"
                className="btn-icon btn-icon-primary btn-icon-md previous page-link"
                disabled={true}
                onClick={[Function]}
                tabIndex="-1"
                type="button"
              >
                <span
                  className="btn-icon__icon-container"
                >
                  <span
                    className="pgn__icon btn-icon__icon"
                  >
                    <svg
                      aria-hidden={true}
                      fill="none"
                      focusable={false}
                      height={24}
                      role="img"
                      viewBox="0 0 24 24"
                      width={24}
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M17.51 3.87 15.73 2.1 5.84 12l9.9 9.9 1.77-1.77L9.38 12l8.13-8.13Z"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                </span>
              </button>
            </li>
            <li
              className="page-item disabled"
            >
              <button
                aria-label="Next, Page 2"
                className="btn-icon btn-icon-primary btn-icon-md next page-link"
                disabled={true}
                onClick={[Function]}
                tabIndex="-1"
                type="button"
              >
                <span
                  className="btn-icon__icon-container"
                >
                  <span
                    className="pgn__icon btn-icon__icon"
                  >
                    <svg
                      aria-hidden={true}
                      fill="none"
                      focusable={false}
                      height={24}
                      role="img"
                      viewBox="0 0 24 24"
                      width={24}
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="m6.49 20.13 1.77 1.77 9.9-9.9-9.9-9.9-1.77 1.77L14.62 12l-8.13 8.13Z"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                </span>
              </button>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </div>
</div>
`;
